StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs

```diff
@@ -365,5 +365,91 @@ public void TestMethod()
 
             await VerifyCSharpFixAsync(testCode, expectedDiagnostics, fixedTestCode, CancellationToken.None).ConfigureAwait(false);
         }
+
+        [Fact]
+        [WorkItem(2888, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2888")]
+        public async Task TestQueryExpressionWithMissingSelectAsync()
+        {
+            var testCode = @"namespace TestNamespace
+{
+    using System.Linq;
+
+    public class TestClass
+    {
+        private int[] testArray = { 1, 2, 3, 4, 5 };
+
+        public void TestMethod()
+        {
+            var x = from element in testArray
+                    where element > 1;
+        }
+    }
+}
+";
+
+            var expectedDiagnostics = new DiagnosticResult("CS0742", DiagnosticSeverity.Error).WithLocation(12, 38).WithMessage("A query body must end with a select clause or a group clause");
+            await VerifyCSharpFixAsync(testCode, expectedDiagnostics, testCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        [WorkItem(2888, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2888")]
+        public async Task TestQueryExpressionWithMissingSelect2Async()
+        {
+            var testCode = @"namespace TestNamespace
+{
+    using System.Linq;
+
+    public class TestClass
+    {
+        private int[] testArray = { 1, 2, 3, 4, 5 };
+
+        public void TestMethod()
+        {
+            var x = from element in testArray where element < 3
+                    where element > 1;
+        }
+    }
+}
+";
+            var fixedTestCode = @"namespace TestNamespace
+{
+    using System.Linq;
+
+    public class TestClass
+    {
+        private int[] testArray = { 1, 2, 3, 4, 5 };
+
+        public void TestMethod()
+        {
+            var x = from element in testArray
+                where element < 3
+                    where element > 1;
+        }
+    }
+}
+";
+
+            await new CSharpTest
+            {
+                TestState =
+                {
+                    Sources = { testCode },
+                    ExpectedDiagnostics =
+                    {
+                        Diagnostic(SA110xQueryClauses.SA1103Descriptor).WithLocation(11, 21),
+                        new DiagnosticResult("CS0742", DiagnosticSeverity.Error).WithLocation(12, 38).WithMessage("A query body must end with a select clause or a group clause"),
+                    },
+                },
+                FixedState =
+                {
+                    Sources = {fixedTestCode },
```

Comment:
Should be `Sources = { fixedTestCode }`, build failed on this.

Reply:
Fixed